system/core/pages.php

```diff
@@ -161,6 +210,21 @@ function newPage() {
 		$html->writeScript(ob_get_clean());
 	}
 
+	/*
+		Function: createPageInstance
+		Creates a page object (appropriate subclass of
+		HyphaDatatypePage, e.g. textpage) instance from the
+		given page node.
+
+		Parameters:
+		RequestContext $O_O
+		HyphaDomElement $node - The page node from $hyphaXml
+	*/
+	function createPageInstance(RequestContext $O_O, HyphaDomElement $node) {
```

Comment:
I observe TypeDefs (which I like). Should we add these to all functions or do we require lower PHP version support?

Reply:
Yeah, we started adding type annotations on arguments at some point, so ideally we add them on all functions we add or change. Maybe an overall refactor to add them everywhere might be good. but also some work.